cmd/capibmadm/cmd/powervs/port/list.go

```diff
@@ -0,0 +1,137 @@
+/*
+ Copyright 2023 The Kubernetes Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package port
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	powerClient "github.com/IBM-Cloud/power-go-client/clients/instance"
+
+	logf "sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/iam"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/powervs"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/printer"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/utils"
+)
+
+var network string
+
+// ListCommand powervs port list command.
+func ListCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List PowerVS ports",
+		Example: `
+ # List ports in PowerVS Network
+ export IBMCLOUD_API_KEY=<api-key>
+ capibmadm powervs port list --instance-id <service-instance-id> --network <network ID or name>`,
+	}
+
+	cmd.Flags().StringVar(&network, "network", "", "Network ID or Name(preference will be given to the ID over Name)")
+	_ = cmd.MarkFlagRequired("network")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if err := listPorts(cmd.Context(), network); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	return cmd
+}
+
+func listPorts(ctx context.Context, network string) error {
+
+	log := logf.Log
+	log.Info("Listing PowerVS ports", "service-instance-id", options.GlobalOptions.ServiceInstanceID, "network", network)
+
+	auth := iam.GetIAMAuth()
+	accountID, _ := utils.GetAccountID(ctx, auth)
+	sess, err := powervs.NewPISession(accountID, options.GlobalOptions.PowerVSZone, options.GlobalOptions.Debug)
+	if err != nil {
+		return err
+	}
+
+	c := powerClient.NewIBMPINetworkClient(ctx, sess, options.GlobalOptions.ServiceInstanceID)
+	networks, err := c.GetAll()
+	if err != nil {
+		return fmt.Errorf("failed to get the networks, err: %v", err)
+	}
+
+	var networkNames, networkIDs []string
```

Comment:
map can be used instead of a list and iterating over each element, because that would use a better DS underneath.

Reply:
Removed map and list. Checking for the variable while iteration itself.